From 1ed3c6672957f2033f217f90a76f85973f1c85c6 Mon Sep 17 00:00:00 2001
From: nekral-guest <nekral-guest@5a98b0ae-9ef6-0310-add3-de5d479b70d7>
Date: Sat, 30 Aug 2008 18:29:55 +0000
Subject: [PATCH] * NEWS: Added support for uclibc.
 	* configure.in, libmisc/copydir.c: futimes() and lutimes() are not
 	standard. Check if they are implemented before using them. Do not
 	set the time of links if lutimes() does not exist, and use
 	utimes() as a replacement for futimes().

git-svn-id: svn://svn.debian.org/pkg-shadow/upstream/trunk@2294 5a98b0ae-9ef6-0310-add3-de5d479b70d7
---
 libmisc/copydir.c |    8 ++++++++
 3 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/libmisc/copydir.c b/libmisc/copydir.c
index abcea4c..b887303 100644
--- a/libmisc/copydir.c
+++ b/libmisc/copydir.c
@@ -431,12 +431,14 @@ static int copy_symlink (const char *src, const char *dst,
 		return -1;
 	}
 
+#ifdef HAVE_LUTIMES
 	/* 2007-10-18: We don't care about
 	 *  exit status of lutimes because
 	 *  it returns ENOSYS on many system
 	 *  - not implemented
 	 */
 	lutimes (dst, mt);
+#endif
 
 	return err;
 }
@@ -548,9 +550,15 @@ static int copy_file (const char *src, const char *dst,
 
 	(void) close (ifd);
 
+#ifdef HAVE_FUTIMES
 	if (futimes (ofd, mt) != 0) {
 		return -1;
 	}
+#else
+	if (utimes(dst, mt) != 0) {
+		return -1;
+	}
+#endif
 
 	if (close (ofd) != 0) {
 		return -1;
-- 
1.6.2

